### PR TITLE
feat: better args formatting

### LINKED ISSUE #143: Strange formatting of long show-set rule
```typ
#show heading.where(level: 2): set heading(numbering: (..n) => [#n.pos().at(1).])
```

with `typstyle -c 80`:

```typ
#show heading.where(level: 2): set heading(numbering: (
  ..n,
) => [#n.pos().at(1).])
```

I think it's much more readable to prefer first breaking by 1 parameter per line:

```typ
#show heading.where(level: 2): set heading(
  numbering: (..n) => [#n.pos().at(1).],
)
```

### PR DESCRIPTION
**Pull request by QuadnucYard**
This PR includes:

- A simple optimization. For verbatim leaf text conversions, we do not need to go through the heavy logic in `trivia`. For verbatim inner text conversions, we only need to borrow the source text.
- A breaking style change. We no longer force every single arg of simple expressions to fold. This may bring more flexibility. With the previously implemented chain formatting (#171), func-calls will not become so weird.
  In the current implementation, only when the only arg is positional or spread, and it is not `FuncCall`, `FieldAccess`, `Unary`, or `Binary`, will it be forced to fold. Otherwise, it will fit regardless of flavor.
  According to the snapshots, this has little effect on width -80 and -120.
  Fixes #143.

Before we truly implement compact formatting as in Rust (we all agree that `pretty` cannot achieve this efficiently), we cannot make the result in every context satisfactory.

### CODE REVIEW

File: src/pretty/mod.rs [hunk 1]
```diff
@@ -163,7 +164,25 @@ impl<'a> PrettyPrinter<'a> {
...
+    }
+
+    /// For leaf only.
+    fn convert_verbatim(&'a self, node: impl AstNode<'a>) -> ArenaDoc<'a> {
+        self.convert_verbatim_untyped(node.to_untyped())
+    }
+
+    /// For leaf only.
+    fn convert_verbatim_untyped(&'a self, node: &'a SyntaxNode) -> ArenaDoc<'a> {
+        self.arena.text(node.text().as_str())
```
Comment (Enter-tainer) [inline src/pretty/mod.rs:185]: maybe add a debug assert here to prevent incorrect usage

File: src/pretty/mod.rs [hunk 2]
```diff
@@ -286,6 +289,7 @@ impl<'a> PrettyPrinter<'a> {
     }
 
     fn convert_raw(&'a self, raw: Raw<'a>) -> ArenaDoc<'a> {
+        println!("raw ");
```
Comment (Enter-tainer) [inline src/pretty/mod.rs:292]: oops

File: tests/snapshots/assets__check_file@cetz-manual.typ-120.snap [hunk 3]
```diff
@@ -28,9 +28,11 @@ snapshot_kind: text
...
 #set par(justify: true)
-#set heading(numbering: (..num) => if num.pos().len() < 4 {
-  numbering("1.1", ..num)
-})
+#set heading(
+  numbering: (..num) => if num.pos().len() < 4 {
+    numbering("1.1", ..num)
+  },
+)
 #show link: set text(blue)
```
Comment (Enter-tainer) [inline tests/snapshots/assets__check_file@cetz-manual.typ-120.snap:36]: cool

File: tests/snapshots/assets__check_file@cetz-tree.typ-40.snap [hunk 4]
```diff
@@ -107,10 +109,12 @@ snapshot_kind: text
...
-        panic("Unsupported content type "
-          + type(content)
-          + "! "
-          + "Provide your own `draw-node` implementation.")
+        panic(
+          "Unsupported content type "
+            + type(content)
+            + "! "
+            + "Provide your own `draw-node` implementation.",
+        )
```
Comment (Enter-tainer) [inline tests/snapshots/assets__check_file@cetz-tree.typ-40.snap:117]: this looks nice. But I guess the trailing comma here is unneeded? I'm not quite sure at this moment

Comment (QuadnucYard) [reply tests/snapshots/assets__check_file@cetz-tree.typ-40.snap:117]: needless, but the args are multiline, thus a trailing comma is added

File: src/pretty/mod.rs (hunk 1 above)
Comment (QuadnucYard) [reply src/pretty/mod.rs:185]: actually, the previously frequently used `trivia` is also based on this assertion.
What I didn't have time to point out in the description is that, compared with `trivia`, this `verbatim` has fewer checks for inline items, and may be more performant.

Comment (QuadnucYard) [reply src/pretty/mod.rs:185]: unfortunately, we could not check whether the node is leaf.

File: src/pretty/func_call.rs [hunk 5]
```diff
@@ -55,48 +56,48 @@ impl<'a> PrettyPrinter<'a> {
...
+                    | Expr::Conditional(_)
+                    | Expr::While(_)
+                    | Expr::For(_)
+                    | Expr::Contextual(_)
+                    | Expr::Raw(_)
+            )
+        }) {
+            fold_style = FoldStyle::Always;
+        }
+
```
Comment (Enter-tainer) [inline src/pretty/func_call.rs]: this makes a lot of sense to me.

File: tests/snapshots/assets__check_file@packages-codly.typ-120.snap [hunk 6]
```diff
@@ -943,13 +943,7 @@ snapshot_kind: text
...
 
   // Get the height of an individual line.
-  let line-height = (
-    measure(
-      raw.line(1, 1, "Hello, World!", [Hello, World!]),
-    ).height
-      + padding.top
-      + padding.bottom
-  )
+  let line-height = measure(raw.line(1, 1, "Hello, World!", [Hello, World!])).height + padding.top + padding.bottom
```
Comment (Enter-tainer) [inline tests/snapshots/assets__check_file@packages-codly.typ-120.snap:947]: This becomes a single line. (which is expected)

**Review by Enter-tainer — APPROVED** [review]
LGTM!

**Pull request merged**